enhancements/insights/insights-operator-runtime.md

```diff
@@ -0,0 +1,532 @@
+
+The workload data is gathered every 12 hours by the Insights Operator.
+Gathering the runtime data will increase the overall executing time of the workload gatherer. The increase must remain reasonable.
+
+In the existing initial implementation of the insights-operator-runtime, it takes __~200 ms to scan a single container__.
+As the inspection is done in parallel on each worker node, its overall duration is impacted by
+the density of the cluster (number of containers per worker node).
+On a fresh OpenShift cluster (with no user workload and only OpenShift own containers), it takes ~15 seconds to scan all containers.
+For a 8-worker node with 800 containers, assuming the containers are spread evenly, it would take around 20 seconds to scan the whole cluster. In practice, it will be a bit longer and capped by the inspection of the worker node with the most containers.
+
+The additional data in the `workloadRuntimeInfoContainer` will increase the size of the insights payload that is archived and sent to Red Hat backends.
+
+The additional data adds __~200 uncompressed bytes per WorkloadContainerShape in the JSON payload__.
+
+This additional payload does not scale with the number of pods but with the number of `WorkloadContainerShape` which is difficult to estimate. There is an upper limit of 8000 pods
+in the Insights operator but that does not give a limit on the number of containers. It is also very likely that some of the pods are replicas and their containers would only count for 1 `WorkloadContainerShape` in their gathered data.
+
+#### UX Risk & Mitigation
+
+The additional runtime info added to the gathered data will be exposed to the user though the archives generated by the Insights operator. As the values of the field are hashed, the user will not be able to determine their corresponding content.
+
+Exposing these data to the user with a clean user interface (eg within console.redhat.com) is out of scope of this proposal.
+
+### Drawbacks
+
+If implemented, Red Hat will document the addition of this new collection. We  introduce the risk that some number of customers who would have or are sending telemetry data via the Insights operator will choose to turn it off because they are uncomfortable sharing this additional data. 
+
+The insights-operator-runtime needs to be continually updated to detect new workloads  that Red Hat wants information about. 
+This can be done iteratively and will require an updated reference to the insights-operator-runtime image in the Insights Operator to take into account the additional detections.
+
+As the insights-operator-runtime is sanitizing the data extracted from the containers, it is not able to retrieve "raw" data (such as processes' name and command line) which limits its ability to detect emerging trends. The insights-operator-runtime must provide fingerprints executables for workloads that Red Hat wants to monitor.
+
+## Design Details
+
+### Open Questions [optional]
```

Comment:
we also need to check with teams responsible for various compliance and certification programs we have for Openshift to make sure extended scope of insights-operator does not conflict with existing certification. One example I was thinking of is FedRAMP certification for ROSA: it seems GovCloud builds of ROSA already have insights and telemetry disabled. However there might be other certifications we have achieved in the past that we need to review

Reply:
@xiormeesh Thanks for the feedback, any new collected data from this RFE will stay in scope of the existing gathered data. Please let me know which teams I can contact to properly review the impact of this RFE to our compliances and certs programs and alleviate any concerns.

Reply:
I amended the RFE and added a section in the drawbacks section:

| Any compliance or certification program that prevents to "call back" Red Hat backends will not report the data (as is already the case).